default_warehouse_from_sale_team/models/account_move.py

```diff
@@ -7,7 +7,11 @@ class AccountMove(models.Model):
     def _search_default_journal(self):
         """If a team is provided and it has a sales journal set, take it as 1st alternative"""
         journal = super()._search_default_journal()
-        team = self.env.context.get("salesteam") or self.team_id or self.env.user.sale_team_id
+        team = (
+            self.env.context.get("salesteam")
+            or (self._cache["team_id"] and self.team_id)
```

Comment:
If already checking the cache, why do you need to read the current value?

Reply:
Because the cached value contains only the ID and not the record.